File: src/baekjoon/큐/P3190.java
package baekjoon.큐;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;

public class P3190 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        int[][] field = new int[N][N];
        int aCnt = Integer.parseInt(br.readLine());
        Queue<int[]> aq = new LinkedList<>();
        StringTokenizer token;
        for(int i = 0; i < aCnt; i++){
            token = new StringTokenizer(br.readLine());
            aq.offer(new int[]{Integer.parseInt(token.nextToken()), Integer.parseInt(token.nextToken())});
        }

        int cCnt = Integer.parseInt(br.readLine());
        int[] cSec = new int[cCnt];
        char[] cArr = new char[cCnt];
        for(int i = 0; i < cCnt; i++){
            token = new StringTokenizer(br.readLine());
            cSec[i] = Integer.parseInt(token.nextToken());
            cArr[i] = token.nextToken().toCharArray()[0];
        }

        int sec = 0, dir = 0;
        int sx = 0, sy = 0;
        int[] dx = {1, 0, -1, 0}; // 우하좌상
        int[] dy = {0, 1, 0, -1};
//        List<>posList
        while(!aq.isEmpty()){

        }

        System.out.println(sec);
    }
}

File: src/baekjoon/큐/P3190.kt
package baekjoon.큐

import java.util.*
import kotlin.collections.ArrayList

val posQueue = LinkedList<Pair<Int, Int>>().apply { offer(Pair(1, 1)) }
var N = 0
val apples = ArrayList<Pair<Int, Int>>()
val cmds = LinkedList<Pair<Int, String>>()
fun main(){
    val br = System.`in`.bufferedReader()
    N = br.readLine().toInt()

    repeat(br.readLine().toInt()){
        val input = br.readLine().split(' ')
        apples.add(Pair(input[1].toInt(), input[0].toInt()))
    }

    repeat(br.readLine().toInt()){
        val input = br.readLine().split(' ')
        cmds.offer(Pair(input[0].toInt(), input[1]))
    }

    var sec = 0
    var dir = 0
    var over = false
    // 우하좌상 순
    val dx = intArrayOf(1, 0, -1, 0)
    val dy = intArrayOf(0, 1, 0, -1)
    while(true){
        dir %= 4
        val nx = posQueue.last().first + dx[dir]
        val ny = posQueue.last().second + dy[dir]
        for(p in posQueue.iterator()){
            if(p.first == nx && p.second == ny){
                sec++
                over = true
                break
            }
        }
        if(over) break
        if(nx > N || ny > N || nx < 1 || ny < 1){
            sec++
            break
        }
        posQueue.offer(Pair(nx, ny))
        var getApple = false
        var appleToRemove: Pair<Int, Int>? = null
        for(i in apples.indices){
            if(apples[i].first == nx && apples[i].second == ny) {
                appleToRemove = apples[i]
                getApple = true
                break
            }
        }
        apples.remove(appleToRemove)
        if(!getApple) posQueue.poll()
        sec++

        if(cmds.isNotEmpty() && cmds.peek().first == sec){
            when(cmds.poll().second){
                "L" -> dir--
                "D" -> dir++
            }
        }
    }
//    println(dir)
    println(sec)
}